clang/include/clang/AST/Type.h

```diff
@@ -6169,9 +6169,13 @@ class HLSLAttributedResourceType : public Type, public llvm::FoldingSetNode {
     // Data gathered from HLSL resource attributes
     llvm::dxil::ResourceClass ResourceClass;
```

Comment:
Do we plan to add something like ResourceKind to HLSLAttributedResourceType?
If we have ResourceKind, we could just add a method isRowAccess and check the ResourceKind instead of add a RowAccess field.

Reply:
We are planning to remove `ResourceKind` and `HLSLResourceAttr` completely and use these individual resource attributes instead. llvm/llvm-project#104862